CopyListWithRandomPointer.java

```diff
@@ -38,3 +38,44 @@ private Node clone(Node node){
         return copy;
     }
 }
+
+//TC: O(n)
+//SC: O(n), please confirm if the space will be O(n) or not, as I am having n extra nodes in my list.
```

Comment:
When you implement N nodes, SC: O(N * space complexity of one node). It's correct.